src/sinks/datadog/metrics/request_builder.rs

```diff
@@ -257,6 +277,7 @@ impl IncrementalRequestBuilder<(DatadogMetricsEndpoint, Vec<Metric>)>
             content_type: endpoint.content_type(),
             finalizers,
             batch_size,
+            raw_bytes: self.raw_bytes,
```

Comment:
Alright, so this is a little tricky.

Incremental encoding is expected to be called, for every input batch, like so:
- `encode_events_incremental` with the input as the entire batch, potentially returning multiple paylods
- for each payload, `build_request` is called to take the payload (and its metadata) and transform it into a bonafide request

Currently, you correctly capture the total number of raw encoded bytes generated by the call to `encode_events_incremental`, but you use that single value as the raw byte size for every payload by taking it from `self.raw_size` every time `build_request` is subsequently called. In other words, if we generate 10 requests, and the sum total of raw encoded bytes is 100MB, we're going to end up reporting 1MB total raw bytes sent.

There's a simple fix here which is to add a new field to the metadata tuple (the associated type definition for this encoder) so that the raw bytes flow together with the payload, finalizers, etc. This way, every payload has only its individual raw size, and you can avoid having to have a field on the encoder itself. By adding it to the metadata, it will be present when calling `build_request`.

Reply:
Yes good spot! I did it that way for traces, but missed it here because I was only looking out for bytes before compression..